pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -241,9 +222,15 @@ func (c *Reconciler) deleteTaskRun(namespace, name string) error {
 }
 
 // makeBuild creates a build from the task, using the task's buildspec.
-func (c *Reconciler) makeBuild(t *v1alpha1.Task, tr *v1alpha1.TaskRun) (*buildv1alpha1.Build, error) {
+func (c *Reconciler) makeBuild(tr *v1alpha1.TaskRun, logger *zap.SugaredLogger) (*buildv1alpha1.Build, error) {
+	// Get related task for taskrun
+	t, err := c.taskLister.Tasks(tr.Namespace).Get(tr.Spec.TaskRef.Name)
+	if err != nil {
+		return nil, err
+	}
+
 	if t.Spec.BuildSpec == nil {
-		return nil, fmt.Errorf("nil BuildSpec")
+		return nil, fmt.Errorf("BuildSpec for task %s is nil", t.Name)
```

Comment:
i'm not sure how the spec ended up being a pointer anyway (might have been my doing :sweat_smile: ), instead of checking for `nil` here should we make it so that the Spec isn't a pointer type (which is [consistent with build anyway](https://github.com/knative/build/blob/master/pkg/apis/build/v1alpha1/build_types.go#L42))

(@pivotal-nader-ziada I think you were pointing this out to me when I was trying to make everything pointers, sorry for turning this around on you!)

Reply:
haha, thats okay as long as we end up with the correct type